python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,170 @@
+
+                # match with gaia_stars
+                with Matcher(gaia_stars["coord_ra"], gaia_stars["coord_dec"]) as m:
+                    idx, i1, i2, d = m.query_knn(
+                        cat_stars["coord_ra"],
+                        cat_stars["coord_dec"],
+                        distance_upper_bound=0.5/3600.0,
+                        return_indices=True,
+                    )
+                cat_stars = cat_stars[i2]
+                cat_stars.add_column(gaia_stars["id"][i1],
+                                     name=self.gaia_reference_class.name + "_id")
+
+                for band in cat_info.bands:
+                    # yaml spline fits are per-band, so loop over bands
+                    # read in spline
+                    filename = os.path.join(lsst.utils.getPackageDir("the_monster"),
+                                            "colorterms/",
+                                            f"{cat_info.name}_to_DES_band_{band}.yaml")
+
+                    colorterm_spline = ColortermSpline.load(filename)
+
+                    # apply colorterms to transform to des mag
+                    band_1, band_2 = cat_info.get_color_bands(band)
+                    orig_flux = cat_stars[cat_info.get_flux_field(band)]
+                    orig_flux_err = cat_stars[cat_info.get_flux_field(band)+'Err']
+                    model_flux = colorterm_spline.apply(
+                        cat_stars[cat_info.get_flux_field(band_1)],
+                        cat_stars[cat_info.get_flux_field(band_2)],
+                        orig_flux,
+                    )
+
+                    # Rescale flux error to keep S/N constant
+                    model_flux_err = model_flux * (orig_flux_err/orig_flux)
+
+                    # Append the modeled mags column to cat_stars
+                    cat_stars.add_column(model_flux,
+                                         name=f"decam_{band}_from_{cat_info.name}_flux")
+                    cat_stars.add_column(model_flux_err,
+                                         name=f"decam_{band}_from_{cat_info.name}_fluxErr")
+
+                if self.write_path_inp is None:
+                    write_path = cat_info.path + '_transformed/'
+                    # The PS1 refcat is coming from the Rubin shared repos, so
+                    # the output can't be in that same place. Explicitly set
+                    # the output path if transforming PS1.
+                    if cat_info.name == 'PS1':
+                        write_path = '/sdf/data/rubin/shared/the_monster/sharded_refcats/ps1_transformed'
+                else:
+                    write_path = self.write_path_inp
+
+                if os.path.exists(write_path) is False:
+                    os.makedirs(write_path)
+                write_path += f"/{htmid}.fits"
+
+                # Save the shard to FITS.
+                cat_stars[outcols].write(write_path, overwrite=True)
```

Comment:
So this will write the shards in astropy table format, right?  But this isn't a refcat format, which is an afw `SimpleCatalog`.  As an intermediate state this may not matter, but (e.g.) the read_stars code expects the afw format.  So I think we need to convert this out, as annoying as it is.

In which case we need to define the full schema, with types, etc.

Reply:
Excellent point -- I've added the portion to write the tables as afw `SimpleCatalog`s.